### ISSUE TITLE
Improve OP payload builder

### ISSUE THREAD

**Issue by mattsse** [paradigmxyz/reth#11040]
### Describe the feature

this https://github.com/paradigmxyz/reth/pull/10904 attempted to restructure the OP payload builder. this now has a few sub functions that are hard to follow and some redundant arguments.

imo this should be reverted and go through review again.

**Comment by 0xKitsune** [comment]
Hello I am happy to take this in a different direction or simply leave `optimism_payload()` unchanged. 

The initial goal was to create smaller, reusuable functions that can be used by custom OP Stack builders rather than reimplementing [optimism_payload()](https://github.com/paradigmxyz/reth/blob/f04fc7c5b966a3db6830ddcc31307bbe125001c4/crates/optimism/payload/src/builder.rs#L122) with small changes.

I agree that a single `optimism_payload()` function is much more readable. Let me know if you have any thoughts on the direction or if you would prefer to leave this unchanged!

**Comment by mattsse** [comment]
yeah definitely want to make this nicer, so please simply reopen revert of #11041 again and we take it from there

**Comment by 0xKitsune** [comment]
Sounds good, I reopened a draft PR for further feedback/revisions.

#11052
